In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import plotly.graph_objects as go

In [2]:
women_data = pd.read_csv('women_1_14_3.csv')
women_data.head()

Unnamed: 0                   name Country  UB_mean_score  UB_std  UB_count  \
0         675           SIMONE BILES     USA          14.26    0.18       7.0   
1         608         REBECA ANDRADE     BRA          14.23    0.56      11.0   
2         668          SHILESE JONES     USA          14.51    0.49      15.0   
3         397         KONNOR MCCLAIN     USA          13.27    0.80       3.0   
4         336  JESSICA ROSE GADIROVA     GBR          13.39    0.44      12.0   

   UB_D_mean_score  UB_D_std  UB_E_mean_score  UB_E_std  ...  VT_D_mean_score  \
0             6.00      0.00             8.26      0.18  ...             5.96   
1             6.11      0.14             8.12      0.47  ...             5.27   
2             6.29      0.08             8.22      0.42  ...             5.00   
3             5.60      0.20             7.67      0.60  ...             5.00   
4             5.58      0.14             7.81      0.41  ...             4.98   

   VT_D_std  VT_E_mean_score  VT_E_std  VT_rank  all_scores_std  \
0      0.42             9.26      0.37      1.0            0.41   
1      0.68             9.40      0.29      2.0            0.78   
2      0.00             9.31      0.16      4.0            0.69   
3      0.00             9.28      0.16      5.0            0.67   
4      0.28             8.98      0.36     19.0            0.56   

   Standardized_Score_Sum  Total_Appearances  apparatus_participated  \
0                   58.71               30.0                       4   
1                   56.60               44.0                       4   
2                   55.82               53.0                       4   
3                   55.68               12.0                       4   
4                   54.67               58.0                       4   

   mean_score_of_all  
0              14.70  
1              14.21  
2              13.96  
3              13.92  
4              13.72  

[5 rows x 40 columns]

## Women's Floor Exercise

### Top 15 Athletes in FX by mean_score (3+ Apperances)

In [27]:
FX_data = women_data[women_data['FX_count']>3]
FX_data=FX_data[['name','Country','FX_mean_score','FX_std','FX_count','FX_D_mean_score','FX_D_std','FX_E_mean_score','FX_E_std']]
FX_data.sort_values(by='FX_mean_score', ascending=False).head(10)

name Country  FX_mean_score  FX_std  FX_count  \
0             SIMONE BILES     USA          14.87    0.32       7.0   
1           REBECA ANDRADE     BRA          14.21    0.30       8.0   
4    JESSICA ROSE GADIROVA     GBR          14.10    0.35      14.0   
43          KALIYA LINCOLN     USA          14.01    0.28       6.0   
7               JADE CAREY     USA          13.80    0.39       8.0   
17          MARTINA MAGGIO     ITA          13.77    0.19       7.0   
6            JORDAN CHILES     USA          13.73    0.34      11.0   
2            SHILESE JONES     USA          13.70    0.42      14.0   
13           KAYLA DICELLO     USA          13.62    0.22       8.0   
119  SABRINA MANECA VOINEA     ROU          13.59    0.18       6.0   

     FX_D_mean_score  FX_D_std  FX_E_mean_score  FX_E_std  
0               6.71      0.11             8.18      0.25  
1               6.06      0.11             8.18      0.24  
4               5.86      0.18             8.26      0.21  
43              5.93      0.32             8.14      0.12  
7               5.96      0.12             7.85      0.28  
17              5.69      0.04             8.12      0.11  
6               5.75      0.14             8.00      0.35  
2               5.62      0.16             8.11      0.28  
13              5.45      0.11             8.17      0.14  
119             5.85      0.21             7.74      0.09

#### The best USA athelete in FX is Simone Biles, scoring 14.87 on average with 0.32 standard deviation. She has a much higher D_score than others and a top-level E_score as well. Other outstanding USA atheletes are Kaliya Lincoln and Jade Carey, ranking 4th and 5th, who may compete for another medal. USA also have 5 more on this list, showing a dominancy in this exercise.

### Distribution of Individual's FX Mean Score

In [4]:
fig = go.Figure(data=[go.Histogram(x=FX_data['FX_mean_score'],nbinsx=20)])
fig.update_layout(
    title='Histogram of FX mean score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### The distribution of FX's score shows that on the top level, there is a huge gap between being good and best. It means that the difference is large enough for top level atheletes to stand out from the others. 

### Distribution of Individual's SD

In [5]:
filter_data = FX_data[FX_data['FX_std']<5]
fig = go.Figure(data=[go.Histogram(x=filter_data['FX_std'],nbinsx=20)])
fig.update_layout(
    title='Histogram of FX SD',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### Most atheletes have a SD between 0.2 and 0.4, with a few of them having it lower than 0.2, showing that this apparatus is generally consistent.

In [6]:
FX_sd_of_sd=filter_data['FX_std'].std()
FX_mean_sd=filter_data['FX_std'].mean()
print(FX_mean_sd.round(2),round(FX_sd_of_sd,2))

0.47 0.29


#### For FX, the mean of sd is 0.45, the sd of sd is 0.31. This shows a most atheletes are stable in FX performances, but there is not a big enough distinction between atheletes.

### Distribution of D_Score

In [7]:
fig = go.Figure(data=[go.Histogram(x=FX_data['FX_D_mean_score'],nbinsx=20)])
fig.update_layout(
    title='Histogram of FX D_Score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### The distribution of D_Score shows that leading atheletes are far above others, especially with Simone Biles who leads by a great margin. D_score can separate scores apart for sure.

### Distribution of FX E_Score

In [8]:
fig = go.Figure(data=[go.Histogram(x=FX_data['FX_E_mean_score'],nbinsx=20)])
fig.update_layout(
    title='Histogram of FX E_Score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### E_score is less distinguishing from this distribution, as many can achieve a high E_score without a huge difference.

### Conclusion

### In general, FX relies quite heavily on D_Score to be separated from the others. USA has a handful of good atheletes that grants them the advantage in this apparatus, especially Simone Biles who has the biggest hope to win gold in this game

## Women's Balance Beam

### Top 15 Athletes in BB by mean_score (3+ Apperances)

In [28]:
BB_data = women_data[women_data['BB_count']>3]
BB_data=BB_data[['name','Country','BB_mean_score','BB_std','BB_count','BB_D_mean_score','BB_D_std','BB_E_mean_score','BB_E_std']]
BB_data.sort_values(by='BB_mean_score', ascending=False).head(10)

name Country  BB_mean_score  BB_std  BB_count  BB_D_mean_score  \
0      SIMONE BILES     USA          14.60    0.22       7.0             6.37   
11        YUSHAN OU     CHN          13.97    0.40       9.0             6.16   
50   QINGYING ZHANG     CHN          13.90    0.71       5.0             6.32   
53     MANA OKAMURA     JPN          13.81    0.30       5.0             5.68   
627  URARA ASHIKAWA     JPN          13.81    0.40       5.0             5.76   
726    SANNE WEVERS     NED          13.60    0.44       5.0             5.72   
38        JIN ZHANG     CHN          13.55    0.45       7.0             5.43   
1    REBECA ANDRADE     BRA          13.50    0.51      10.0             5.89   
5      SKYE BLAKELY     USA          13.46    0.67      11.0             6.07   
10   FLAVIA SARAIVA     BRA          13.45    0.87       8.0             5.59   

     BB_D_std  BB_E_mean_score  BB_E_std  
0        0.18             8.23      0.13  
11       0.19             7.82      0.33  
50       0.16             7.58      0.61  
53       0.11             8.13      0.23  
627      0.17             8.05      0.24  
726      0.25             7.88      0.49  
38       0.19             8.12      0.29  
1        0.13             7.62      0.46  
5        0.17             7.39      0.62  
10       0.17             7.87      0.79

#### The best USA athlete in BB is Simone Biles (again), scoring 14.6 on average and leading the second place Yushan Ou by 0.63. She performs stably with a low sd. What truly sets her apart is her outstanding E_score of 8.23, although her D_score of 6.37 is also the highest on chart.

### Distribution of Individual's BB Mean Score

In [10]:
fig = go.Figure(data=[go.Histogram(x=BB_data['BB_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of BB mean score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### There are more competition at the higher level in BB, but the NO.1 Simone Biles is still leading by a big margin. It means that the competition for medals will be intense.

### Distribution of Individual's HB SD

In [11]:
fig = go.Figure(data=[go.Histogram(x=BB_data['BB_std'],nbinsx=25)])
fig.update_layout(
    title='Histogram of BB SD',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### Performance on HB is generally less stable, as fewer atehletes are able to achieve a SD of 0.3 or lower. There is a greater mean of SD as well.

In [12]:
BB_sd_of_sd=BB_data['BB_std'].std()
BB_mean_sd=BB_data['BB_std'].mean()
print(BB_mean_sd.round(2),round(BB_sd_of_sd,2))

0.72 0.3


### Distribution of BB D_Score

In [13]:
fig = go.Figure(data=[go.Histogram(x=BB_data['BB_D_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of BB D_score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### D_score is actually very distinguishing in BB, as only a few atheletes can appear on the top level. But there are more than one in the highest bin.

### Distribution of BB E_Score

In [14]:
fig = go.Figure(data=[go.Histogram(x=BB_data['BB_E_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of BB E_score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### E_Score for BB is very distinguishing as well, as only 4 atheletes score above 8, leading by a lot comparing to others.
### Conclusion
### For BB, USA has the best athelete in the world Simone Biles who would also likely win a gold medal, but they don't have as many contenders for other medals as FX. Both D_score and E_score are crucial in setting good atheletes apart from others.
## Women's Uneven Bars

### Top 15 Athletes in UB by mean_score (3+ Apperances)

In [29]:
UB_data = women_data[women_data['UB_count']>3]
UB_data=UB_data[['name','Country','UB_mean_score','UB_std','UB_count','UB_D_mean_score','UB_D_std','UB_E_mean_score','UB_E_std']]
UB_data.sort_values(by='UB_mean_score', ascending=False).head(10)

name Country  UB_mean_score  UB_std  UB_count  \
32      XIAOYUAN WEI     CHN          14.54    0.69       6.0   
2      SHILESE JONES     USA          14.51    0.49      15.0   
16        QIYUAN QIU     CHN          14.49    1.03      10.0   
0       SIMONE BILES     USA          14.26    0.18       7.0   
1     REBECA ANDRADE     BRA          14.23    0.56      11.0   
8      ALICE D AMATO     ITA          14.20    0.58      16.0   
57    REBECCA DOWNIE     GBR          14.16    0.32       5.0   
621    GIORGIA VILLA     ITA          14.11    0.10      11.0   
499  ELISABETH SEITZ     GER          14.05    0.75       9.0   
56        ZOE MILLER     USA          14.01    1.26       9.0   

     UB_D_mean_score  UB_D_std  UB_E_mean_score  UB_E_std  
32              6.37      0.29             8.17      0.43  
2               6.29      0.08             8.22      0.42  
16              6.43      0.23             8.06      0.93  
0               6.00      0.00             8.26      0.18  
1               6.11      0.14             8.12      0.47  
8               6.16      0.10             8.03      0.50  
57              6.02      0.27             8.14      0.25  
621             5.81      0.03             8.30      0.12  
499             6.01      0.14             8.04      0.63  
56              6.24      0.37             7.80      0.83

#### The best USA athelete in UB is Shilese Jones, ranking 2nd falling slightly short behind Xiaoyuan Wei from China. The score margin is only 0.03, and Shilese has a more stable performance than Xiaoyuan. The competition for gold will be intense and undecided. Also, Simone Biles ranks 4th in UB, not likely to win gold but may contend for a medal.
### Distribution of UB Mean Score

In [16]:
fig = go.Figure(data=[go.Histogram(x=UB_data['UB_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of UB mean score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### The distribution of mean score is very clustered on the top level, meaning that there will be a very intense competition for medals. 
### Distribution of PB SD

In [17]:
filter_data_UB = UB_data[UB_data['UB_std']<3]
fig = go.Figure(data=[go.Histogram(x=filter_data_UB['UB_std'],nbinsx=30)])
fig.update_layout(
    title='Histogram of UB SD',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### Few atheletes are able to have a low sd, so consistency is rather low and unstable in UB. Also, the mean of sd is high.

In [18]:
#Get rid of outliers
filter_data_UB = UB_data[UB_data['UB_std']<2]
UB_sd_of_sd=filter_data_UB['UB_std'].std()
UB_mean_sd=filter_data_UB['UB_std'].mean()
print(UB_mean_sd.round(2),round(UB_sd_of_sd,2))

0.7 0.37


### Distribution of UB D_Score

In [19]:
fig = go.Figure(data=[go.Histogram(x=UB_data['UB_D_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of UB D_score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### UB's D_score is very distinguishing on the upper level, meaning that it's important to attempt a high D_score.
### Distribution of UB's E_Score

In [20]:
fig = go.Figure(data=[go.Histogram(x=UB_data['UB_E_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of UB E_score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### E_score is less distinguishing for UB, as many are able to achieve a high score above 8.
### Conclusion
### UB has very intense competition for medals. USA has one athelete ranked 2nd and competing for gold, and one ranked 4th competing for medal. Performance is generally unstable in UB, and D_score is more important than E_score in higher level.

## Women's Vault
### Top 15 Athletes in VT by mean_score (5+ Apperances)

In [30]:
VT_data = women_data[women_data['VT_count']>5]
VT_data=VT_data[['name','Country','VT_mean_score','VT_std','VT_count','VT_D_mean_score','VT_D_std','VT_E_mean_score','VT_E_std']]
VT_data.sort_values(by='VT_mean_score', ascending=False).head(10)

name Country  VT_mean_score  VT_std  VT_count  \
0         SIMONE BILES     USA          14.98    0.41       9.0   
1       REBECA ANDRADE     BRA          14.66    0.93      15.0   
7           JADE CAREY     USA          14.50    0.28      15.0   
2        SHILESE JONES     USA          14.29    0.18      11.0   
6        JORDAN CHILES     USA          14.22    0.22      19.0   
5         SKYE BLAKELY     USA          14.15    0.20       6.0   
20    ONDINE ACHAMPONG     GBR          14.15    0.20      11.0   
25  TIANA SUMANASEKERA     USA          14.12    0.11       6.0   
47        SEOJEONG YEO     KOR          14.09    0.48      17.0   
8        ALICE D AMATO     ITA          14.06    0.18      10.0   

    VT_D_mean_score  VT_D_std  VT_E_mean_score  VT_E_std  
0              5.96      0.42             9.26      0.37  
1              5.27      0.68             9.40      0.29  
7              5.44      0.27             9.07      0.23  
2              5.00      0.00             9.31      0.16  
6              4.94      0.10             9.29      0.15  
5              5.00      0.00             9.17      0.17  
20             5.00      0.00             9.15      0.20  
25             5.00      0.00             9.15      0.07  
47             5.16      0.32             8.94      0.42  
8              5.00      0.00             9.06      0.18

#### Four out of top 5 atheletes in women's VT come from USA. Simone Biles lead by 0.32 at the first place, who has a much higher D_score than any other athelete. USA is incredibly strong in VT.
### Distribution of Women's VT mean score

In [22]:
fig = go.Figure(data=[go.Histogram(x=VT_data['VT_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of VT mean score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### The score difference is rather large at higher level for VT, and few atheletes are in that top level.
### Distribution of Women's VT SD

In [23]:
fig = go.Figure(data=[go.Histogram(x=VT_data['VT_std'],nbinsx=25)])
fig.update_layout(
    title='Histogram of VT SD',
    xaxis_title='Score',
    yaxis_title='Count'
)

In [24]:
#Get rid of outliers
filter_data_VT = VT_data[VT_data['VT_std']<2]
VT_sd_of_sd=filter_data_VT['VT_std'].std()
VT_mean_sd=filter_data_VT['VT_std'].mean()
print(VT_mean_sd.round(2),round(VT_sd_of_sd,2))

0.42 0.21


#### A handful of atheletes are able to have a low SD, and the mean of SD is also quite low, meaning that VT is a consistent apparatus for atheletes to compete in.

In [25]:
fig = go.Figure(data=[go.Histogram(x=VT_data['VT_D_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of VT D_score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### The D_Score difference in VT is huge in VT, as the first place Simone Biles leads by at least 0.5, and 5 other athelets lead others by about 0.2. It can set atheletes apart easily.

In [26]:
fig = go.Figure(data=[go.Histogram(x=VT_data['VT_E_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of VT E_score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### E_score is also distinguishing in VT, as only a few are able to achieve high E_score above 9.2
### Conclusion
### USA has a huge advantage in Women's VT, as it has many elite atheletes and Simone Biles leading by a large margin. The distinction is rather big on the elite level, and both D_score and E_score are important. Performance is generally stable.